File: Quize3/app/src/main/java/com/example/quize3/MainActivity.java
package com.example.quize3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {

    private Spinner spinner;
    private static final String[] paths = {"Cairo", "London", "Abu Dhabi"};
    String city="Cairo";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        spinner = (Spinner)findViewById(R.id.spinner);
        ArrayAdapter<String>adapter = new ArrayAdapter<String>(MainActivity.this,
                android.R.layout.simple_spinner_item,paths);

        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);
        spinner.setOnItemSelectedListener(this);

    }
    public void checkWeather(android.view.View view) {



        TextView tvCiy=findViewById(R.id.tv_city);

        if(city!=null&&tvCiy.getText().toString().trim().equals("")){

           return;
        }

        tvCiy.setText(city);
        Intent intent=new Intent(this,Detail_activity.class);
        intent.putExtra("city",city);
        startActivity(intent);
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case 0:
               city="Cairo";
                break;
            case 1:
                city="London";

                break;
            case 2:
                city="Abu Dhabi";
                break;

        }
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }
}

File: Lab7/app/src/main/java/com/example/lab7/MainActivity.kt
package com.example.lab7

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.lab7.databinding.MainActivityBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding:MainActivityBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding= MainActivityBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnNextScreen.setOnClickListener {
            if(binding.etName.text.trim().toString().isNullOrEmpty()){
                binding.etName.error="Name can't be empty."
            }else{
                startActivity(Intent(this@MainActivity,SecondActivity::class.java).apply {
                    putExtra("name",binding.etName.text.toString())
                })
            }
        }


    }
}